msu/systems/serialization/strict_deserialization_emulator.nut

```diff
@@ -0,0 +1,66 @@
+// emulates the _in object passed to onDeserialize functions
+::MSU.Class.StrictDeserializationEmulator <- class extends ::MSU.Class.StrictSerDeEmulator
```

Comment:
I could not understand fully the need for these new ser/de emulator classes and would like to find a way to use the original ones we already have. But my brain is rather slow atm and it seems this isn't the main point of the PR so I'll talk to you on VC about this - don't want to delay my review for this.